mne/preprocessing/_beer_lambert_law.py

```diff
@@ -0,0 +1,136 @@
+# Authors: Robert Luke <mail@robertluke.net>
+#          Eric Larson <larson.eric.d@gmail.com>
+#          Alexandre Gramfort <alexandre.gramfort@inria.fr>
+#
+# License: BSD (3-clause)
+
+import os.path as op
+
+import re as re
+import numpy as np
+
+from .. import rename_channels
+from ..io import BaseRaw
+from ..io.pick import _picks_to_idx
+from ..io.constants import FIFF
+from ..utils import _validate_type
+
+
+def beer_lambert_law(raw, ppf=0.1):
+    r"""Convert NIRS optical density data to haemoglobin concentration.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The optical density data.
+    ppf : float
+        The partial pathlength factor.
+
+    Returns
+    -------
+    raw : instance of Raw
+        The modified raw instance.
+    """
+    raw = raw.copy().load_data()
+    _validate_type(raw, BaseRaw, 'raw')
+
+    freqs = np.unique(_channel_frequencies(raw))
+    picks = _check_channels_ordered(raw, freqs)
+    abs_coef = _load_absorption(freqs)
+    distances = _probe_distances(raw)
+
+    for ii in picks[::2]:
+
+        EL = abs_coef * distances[ii] * ppf
+        iEL = np.linalg.pinv(EL)
```

Comment:
we use scipy.linalg usually. So `from scipy import linalg` and `linalg.pinv`

Reply:
Thanks